packages/apps/kadena-docs/src/pages/api/search.ts

```diff
@@ -0,0 +1,43 @@
+import type { MetaData } from '@7-docs/edge';
+import { getCompletionHandler, pinecone } from '@7-docs/edge';
+
+const namespace = 'kda-docs';
+
+export const prompt = `Context: {CONTEXT}
+
+Question: {QUERY}
+
+Answer:`;
+
+export const system = `You are an enthusiastic expert on the subject of kadena and eager to help out!
+Answer the question faithfully using the provided context.
+Use Markdown.
+Always try to include a code example in language-specific fenced code blocks, preferably typescript or pact, especially if it's provided in the context.
+If the answer is not provided in the context, say "Sorry, I don\'t have that information.".
+Return the used documents in metadata`;
+
+type QueryFn = (vector: number[]) => Promise<MetaData[]>;
+
+const OPENAI_API_KEY = process.env.OPENAI_API_KEY;
+const PINECONE_URL = process.env.PINECONE_URL;
+const PINECONE_API_KEY = process.env.PINECONE_API_KEY;
+
+if (OPENAI_API_KEY === undefined)
```

Comment:
Maybe better to use `!OPENAI_API_KEY` to also throw for empty strings (I think in `.env` files this is the case when `KEY=` is set without value).

Reply:
I wish I could. But because of rules I need to be explicit.
It is not a boolean and `Boolean(OPENAI_API_KEY)` is not allowed by the rules

Reply:
I detest that we are pushing code we don't intent because of those rules. We don't need to pretend this is some mission-critical flow. It should be fun to code apps without fighting the rules and waste our time.

Another argument to move applications/code like this out of the monorepo, let's try to accelerate that.